custom/icds_reports/reports/disha.py

```diff
@@ -29,7 +30,9 @@ def __init__(self, state_name, month):
         self.month = month
 
     def _blob_id(self):
-        return 'disha_dump-{}-{}.json'.format(self.state_name.encode('ascii', 'ignore').replace(" ", ""), self.month.strftime('%Y-%m-%d'))
+        # strip all non-alphanumeric chars
+        safe_state_name = re.sub('[^0-9a-zA-Z]+', '', self.state_name)
```

Comment:
Curious, what state name is even causing this? I thought we already have a list of possible state names that can be passed, right? if we just check with those we would not have this issue?

Reply:
This was due to `&` in `Daman & Diu`. Earlier I was just taking care of unicode, but this is real.

I will just convert this code latter to use state-id instead of name.

Reply:
> This was due to & in Daman & Diu
Cool.

I like state name, its better to use that in case the id is changed if env changes/we move out to TCL ;) ?